pypeit/core/coadd.py

```diff
@@ -2738,9 +2747,32 @@ def ech_combspec(waves_arr_setup, fluxes_arr_setup, ivars_arr_setup, gpms_arr_se
         for iord in range(norders[isetup]):
             ind_start = iord*nexps[isetup]
             ind_end = (iord+1)*nexps[isetup]
+            wave_grid_ord = waves_setup_list[isetup][ind_start]
+            wave_grid_diff_ord = np.diff(wave_grid_ord)
+            wave_grid_diff_ord = np.append(wave_grid_diff_ord, wave_grid_diff_ord[-1])
+            wave_grid_mid_ord = wave_grid_ord + wave_grid_diff_ord / 2.0
+            # removing the last bin since the midpoint now falls outside of wave_grid rightmost bin. This matches
+            # the convention in wavegrid above
+            wave_grid_mid_ord = wave_grid_mid_ord[:-1]
+            #trim off first and last pixel in case of edge effects in wavelength calibration
+            #wave_grid_mid_ord = wave_grid_mid_ord[1:-1]
+            #wave_grid_ord = wave_grid_ord[1:-1]
+            # if the wavelength grid is non-monotonic, resample onto a loglam grid
+            if np.any(wave_grid_diff_ord < 0):
+                msgs.warn(f'This order ({iord}) has a non-monotonic wavelength solution. Resampling now: ')
+                #_, dloglam_data, _, _ = wvutils.get_sampling(wave_grid_ord)
+                #wave_grid_ord, wave_grid_mid_ord, _ = wvutils.wavegrid(np.min(wave_grid_ord), np.max(wave_grid_ord), dloglam_data,
+                #                    spec_samp_fact=spec_samp_fact, log10=True)
+                wave_grid_ord = np.linspace(np.min(wave_grid_ord), np.max(wave_grid_ord), len(wave_grid_ord))
+                wave_grid_diff_ord = np.diff(wave_grid_ord)
+                wave_grid_diff_ord = np.append(wave_grid_diff_ord, wave_grid_diff_ord[-1])
+                wave_grid_mid_ord = wave_grid_ord + wave_grid_diff_ord / 2.0
+                # removing the last bin since the midpoint now falls outside of wave_grid rightmost bin. This matches
+                # the convention in wavegrid above
+                wave_grid_mid_ord = wave_grid_mid_ord[:-1]
```

Comment:
Three things here:

1. I think you can trim this because there are a number of repeated lines.  E.g., I think you can start with:
```
wave_grid_ord = waves_setup_list[isetup][ind_start]
if np.any(np.diff(wave_grid_ord) < 0):
    msgs.warn(f'This order ({iord}) has a non-monotonic wavelength solution. Resampling.')
    wave_grid_ord = np.linspace(np.min(wave_grid_ord), np.max(wave_grid_ord), len(wave_grid_ord))
wave_grid_diff_ord = np.diff(wave_grid_ord)
...
```
That should limit the repeated lines.

2. I'm unsure whether you need the `append` line, given that you remove the last element later.

3. I would remove any commented code.